spec/dummy_app/config/initializers/rails_admin.rb

```diff
@@ -2,8 +2,8 @@
 
 RailsAdmin.config do |c|
   c.asset_source = CI_ASSET
-  c.model 'Team' do
```

Comment:
I think this was where I got tripped up in my config (using model constants in the initializer), leading to https://github.com/railsadminteam/rails_admin/issues/3534

It feels problematic to continue to allow this (and makes sense to me that one shouldn't reference model _constants_ in an initializer, https://github.com/railsadminteam/rails_admin/issues/3542 highlights another possibly related issue). If need be, the config block for a model can already live in the model itself (which helps keep the initializer more manageable in large applications).

I'd be in favor of removing support for using model constants in the initializer entirely (given that using the model name as a String continues to work as expected), particularly if that can help avoid needing to override `const_missing`

Reply:
Got it, I think it works well given the cases needed to be handled here - just always cautious with overrides